src/key.rs

```diff
@@ -268,54 +267,87 @@ where
     values: Vec<Option<RedisString>>,
 }
 
-impl<'a, T> HMGetResult<'a, T>
+pub struct HMGetIter<'a, T>
+where
+    T: Into<Vec<u8>>,
+{
+    ai: std::slice::Iter<'a, T>,
+    bi: std::vec::IntoIter<Option<RedisString>>,
```

Comment:
The names `ai` and `bi` aren't very clear. I suggest calling them something like `field_iter` and `value_iter`.

Reply:
Fixed in latest commit.